src/Orc.DataAccess/Database/Models/DbProviderInfo.cs

```diff
@@ -1,20 +1,35 @@
 ﻿namespace Orc.DataAccess.Database
 {
+    using System;
+
     public class DbProviderInfo
     {
-        public string Name { get; set; }
-        public string InvariantName { get; set; }
-        public string Description { get; set; }
-        public string AssemblyQualifiedName { get; set; }
+        public DbProviderInfo(string name, string invariantName, string description, string assemblyQualifiedName)
```

Comment:
I think another alternative is to set it all to string.Empty and still allow object initializers.

Reply:
Note: we decided to keep it as is. Feels like this API with null-checks on ctor makes you always use this class in correct way.